mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/resourcecluster/TaskExecutorState.java

```diff
@@ -259,6 +259,13 @@ protected TaskExecutorGateway getGateway() throws ExecutionException, Interrupte
         return this.gateway.get();
     }
 
+    protected CompletableFuture<TaskExecutorGateway> getGatewayAsync() {
```

Comment:
nit: no need to explicitly call this method async since the type already reflects that.

Reply:
Also, get rid of the blocking method.